lib/pages/class_feedback/view/class_feedback_view.dart

```diff
@@ -0,0 +1,689 @@
+import 'package:acs_upb_mobile/widgets/toast.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_feather_icons/flutter_feather_icons.dart';
+import 'package:provider/provider.dart';
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:recase/recase.dart';
+
+class ClassFeedbackView extends StatefulWidget {
+  const ClassFeedbackView({Key key, this.classHeader}) : super(key: key);
+
+  final ClassHeader classHeader;
+
+  @override
+  _ClassFeedbackViewState createState() => _ClassFeedbackViewState();
+}
+
+class _ClassFeedbackViewState extends State<ClassFeedbackView> {
+  final formKey = GlobalKey<FormState>();
+  TextEditingController classController;
+  bool agreedToResponsibilities;
+
+  List<String> involvementPercentages = [];
+  String selectedTeacherName;
+  Person selectedAssistant;
+  List<Person> classTeachers = [];
+  Map<String, dynamic> feedbackQuestions = {};
+  List<String> responses = [];
+  List<Map<int, bool>> initialValues = [];
+
+  @override
+  void initState() {
+    super.initState();
+
+    agreedToResponsibilities = false;
+    classController = TextEditingController(text: widget.classHeader?.id ?? '');
+    involvementPercentages = [
+      '0% ... 20%',
+      '20% ... 40%',
+      '40% ... 60%',
+      '60% ... 80%',
+      '80% ... 100%'
+    ];
+    Provider.of<PersonProvider>(context, listen: false)
+        .fetchPeople()
+        .then((teachers) => setState(() => classTeachers = teachers));
+
+    awaitFeedbackQuestions();
+  }
+
+  Future<Map<String, dynamic>> awaitFeedbackQuestions() async {
+    await Provider.of<FeedbackProvider>(context, listen: false)
+        .fetchQuestions()
+        .then((questions) => setState(() => feedbackQuestions = questions));
+
+    for (int i = 0; i <= feedbackQuestions.length; i++) {
+      responses.insert(i, '-1');
+      initialValues.insert(i, {
+        0: false,
+        1: false,
+        2: false,
+        3: false,
+        4: false,
+      });
+    }
+
+    return feedbackQuestions;
+  }
+
+  Widget autocompleteAssistant(BuildContext context) {
+    return Autocomplete<Person>(
+      key: const Key('AutocompleteAssistant'),
+      fieldViewBuilder: (BuildContext context,
+          TextEditingController textEditingController,
+          FocusNode focusNode,
+          VoidCallback onFieldSubmitted) {
+        textEditingController.text = selectedAssistant?.name;
+        return TextFormField(
+          controller: textEditingController,
+          decoration: InputDecoration(
+            labelText: S.of(context).labelAssistant,
+            prefixIcon: const Icon(FeatherIcons.user),
+          ),
+          focusNode: focusNode,
+          onFieldSubmitted: (String value) {
+            onFieldSubmitted();
+          },
+          validator: (_) {
+            if (textEditingController.text.isEmpty ?? true) {
+              return S.current.warningYouNeedToSelectAtLeastOne;
```

Comment:
This warning doesn't make any sense here, it's for things where you need to select something from a list. It's not like you can actually select more than one assistant... Just make a custom error message.